overlord/snapstate/snapstate.go

```diff
@@ -2283,6 +2283,11 @@ func AddLinkNewBaseOrKernel(st *state.State, ts *state.TaskSet) (*state.TaskSet,
 	linkSnap.Set("snap-setup-task", snapSetupTask.ID())
 	linkSnap.WaitFor(prev)
 	ts.AddTask(linkSnap)
+	// make sure that remodel can identify which tasks introduce actual
+	// changes to the system and order them correctly
+	if edgeTask := ts.MaybeEdge(DownloadAndChecksDoneEdge); edgeTask == nil {
+		ts.MarkEdge(allTasks[len(allTasks)-1], DownloadAndChecksDoneEdge)
+	}
```

Comment:
I don't understand this bit, partly because we seem to be "pushing" download-and-checks-done edge further to link-snap (the name of the edge doesn't quite fit anymore, but we cannot easily change it, so it's fine). Two questions:
1. Can we set the edge on the previous task?
2. Can we set the edge using a specific task by the name of the variable to be more explicit (e.g. in this case `ts.MarkEdge(linkSnap, DownloadAndChecksDoneEdge)`?

Reply:
allTasks is not ts, so this is not the added linkSnap, otoh the reuse of the edge name might be problematic for other reasons, see my other question,  maybe we need a different name and the code in remodel to differentiate two cases?